### PR TITLE
Phase2-hgx105 Update DetId class for new HGCal module

### PR DESCRIPTION
**Pull request by bsunanda**
Update DetId of HGCalScintillator and introduce for HGCal Trigger

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: DataFormats/ForwardDetId/src/HGCalTriggerDetId.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  return (*this);
+}
+
+int HGCalTriggerDetId::triggerCellX() const {
+  int nT = (type() == HGCSiliconDetId::HGCalFine) ? 
+    HGCSiliconDetId::HGCalFineTrigger : HGCSiliconDetId::HGCalCoarseTrigger;
+  int N  = nT*HGCalTriggerCell;
+  std::vector<int> vc = cellV();
+  int x(0);
+  for (auto const & v : vc)
```
Comment (civanch) [inline DataFormats/ForwardDetId/src/HGCalTriggerDetId.cc]: @bsunanda , here and below I would use brackets to make for loops more visible. Is there any chance that cellV() may be zero length? Likely not.

Comment (bsunanda) [reply DataFormats/ForwardDetId/src/HGCalTriggerDetId.cc]: I don't think so

**Comment by davidlange6** [comment]
please test

**Comment by kpedro88** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**